whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java

```diff
@@ -45,35 +48,100 @@ public BouncedMessageParser(@Value("${mail.smtp.from:}") final String smtpFrom)
 
     @Nullable
     public EmailMessageInfo parse(final MimeMessage message) throws MessagingException, MailParsingException {
-        if (!enabled || !isMultipartReport(message) ){
+        if (!enabled) {
             return null;
         }
 
-        try {
-            final MultipartReport multipartReport = multipartReport(message.getContent());
-            if (isReportDeliveryStatus(multipartReport)) {
-                final DeliveryStatus deliveryStatus = deliveryStatus(message);
-                if (isFailed(deliveryStatus)) {
-                    final MimeMessage returnedMessage = multipartReport.getReturnedMessage();
-                    final String messageId = getMessageId(returnedMessage.getMessageID());
-                    final List<String> recipient = extractRecipients(deliveryStatus);
-                    return new EmailMessageInfo(recipient, messageId);
+        if (isMultipartReport(message)) {
+            try {
+                final MultipartReport multipartReport = multipartReport(message.getContent());
+                if (isReportDeliveryStatus(multipartReport)) {
+                    final DeliveryStatus deliveryStatus = deliveryStatus(message);
+                    if (isFailed(deliveryStatus)) {
+                        final MimeMessage returnedMessage = getReturnedMessage(multipartReport);
+                        final String messageId = getMessageId(returnedMessage.getMessageID());
+                        final List<String> recipient = extractRecipients(deliveryStatus);
+                        return new EmailMessageInfo(recipient, messageId);
+                    }
+                }
+            } catch (MessagingException | IOException | IllegalStateException ex){
+                throw new MailParsingException("Error parsing multipart report");
+            }
+            // multipart report can *only* be a failure
+            throw new MailParsingException("MultiPart message without failure report");
+        }
+
+        if (isMultipartMixed(message)) {
```

Comment:
In this case. Shouldn't we return a MailParsingException??
I mean, should multipartMixed be processed as an error?? as we do with multipart report??

The mail updates are always in text plain, right? So everything that is not a text plain should not go to mail updates

Reply:
The mail updates are always in text plain, right? -> NO